In [5]:
import pandas as pd
from google.colab import files
import io

upload_files = files.upload()
for filename in upload_files.keys():
  data = pd.read_csv(io.StringIO(upload_files[filename].decode('utf-8')), header=None)

print(data.head())

Saving diamonds-1.csv to diamonds-1.csv
    0      1        2      3        4      5      6      7     8     9     10
0  NaN  carat      cut  color  clarity  depth  table  price     x     y     z
1  1.0   0.23    Ideal      E      SI2   61.5     55    326  3.95  3.98  2.43
2  2.0   0.21  Premium      E      SI1   59.8     61    326  3.89  3.84  2.31
3  3.0   0.23     Good      E      VS1   56.9     65    327  4.05  4.07  2.31
4  4.0   0.29  Premium      I      VS2   62.4     58    334   4.2  4.23  2.63


<h2>Quiz 1 : Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [2]:
# Membuat dataframes dari dictionary
obj_dict = {'id':[101, 102, 103, 104, 105], 
            'nama':['Muhammad Fikrie', 'Muhammad Yahya', 'Muhammad Dzaki', 'Muhammad Irul', 'Tita Alfiani'],
            'tahun_lahir':['1999', '1995', '1997', '2003', '2000']}

df = pd.DataFrame(obj_dict)
print(df)

    id             nama tahun_lahir
0  101  Muhammad Fikrie        1999
1  102   Muhammad Yahya        1995
2  103   Muhammad Dzaki        1997
3  104    Muhammad Irul        2003
4  105     Tita Alfiani        2000


<h2>Quiz 2 : Membaca data dari file</h2>

Download file bernama diamonds.csv, kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [6]:
# Membaca data dan memasukannya kedalam suatu variabel
df = pd.read_csv('diamonds-1.csv')
# df.describe()
print(df)

       Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]


Expected Result :

![image.png](attachment:image.png)

<h2>Quiz 3 : DataFrame vs Series</h2>

- Apa perbedaan dataframe dan series
- Buatlah suatu data series

# Perbedaan DataFrame vs Series
- Series adalah suatu objek satu dimensi yang dapat menyimpan berbagai jenis tipe data seperti integer, string, dan lain sebagainya. tipe data dari objek series ini harus seragam.
- DataFrame adalah suatu objek 2 dimensi tempat menyimpan data dengan lebih terstruktur. dataframe memiliki 2 index, yaitu index baris dan index columns. Dalam satu column dataframe harus memiliki tipe data yang sama. tapi antar columnnya dataframe bisa memiliki jenis data yang berbeda.

In [7]:
# code here
# Contoh code DataFrame
print("====Ini Contoh DataFrame====")
df = pd.DataFrame({'tipe_int':[1, 2, 3], 'tipe_str':['Budi', 'Ana', 'Ani']})
print(df)

# Contoh code Series
print("===Ini Contoh Series===")
x = pd.Series([101, 102, 103, 104]) #tanpa menggunakan index, nomor urut di mulai dari 0
print(x)

x.index = ['a', 'b', 'c', 'd'] # var.index digunakan untuk menampilkan no urut
print(x)

#mengagbungkan 2 cara di atas
x = pd.Series([111, 112, 113, 114], index=[1, 2, 3, 4])
print(x)

====Ini Contoh DataFrame====
   tipe_int tipe_str
0         1     Budi
1         2      Ana
2         3      Ani
===Ini Contoh Series===
0    101
1    102
2    103
3    104
dtype: int64
a    101
b    102
c    103
d    104
dtype: int64
1    111
2    112
3    113
4    114
dtype: int64


<h2>Quiz 4 : Pemeriksaan Data Sederhana</h2>

Di quiz no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari :
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [12]:
# code here
# Mengambil beberapa Data pertama
df = pd.read_csv('diamonds-1.csv')
df.head()

Unnamed: 0  carat      cut color clarity  ...  table  price     x     y     z
0           1   0.23    Ideal     E     SI2  ...   55.0    326  3.95  3.98  2.43
1           2   0.21  Premium     E     SI1  ...   61.0    326  3.89  3.84  2.31
2           3   0.23     Good     E     VS1  ...   65.0    327  4.05  4.07  2.31
3           4   0.29  Premium     I     VS2  ...   58.0    334  4.20  4.23  2.63
4           5   0.31     Good     J     SI2  ...   58.0    335  4.34  4.35  2.75

[5 rows x 11 columns]

In [9]:
# Mengambil Nilai dari baris pertama dari column price
print("Nilai dari baris pertama dari column price adalah : ", df.loc[0, 'price'])

Nilai dari baris pertama dari column price adalah :  326


In [10]:
# Mengambil beberapa baris terakhir
df.tail()

Unnamed: 0  carat        cut color  ... price     x     y     z
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[5 rows x 11 columns]

In [13]:
# Mengambil nilai dari baris terakhir di column color
print("Nilai dari baris terakhir dari column color adalah : ", df.loc[53939, 'color'])

Nilai dari baris terakhir dari column color adalah :  D


<h2>Quiz 5 : Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

- Statistik deskriptif adalah metode-metode yang berkaitan dengan pengumpulan dan Penyajian suatu gugus data sehingga memberikan informasi yang berguna.
- Statistik deskriptif adalah metode yang sangat sederhana. Metode ini hanya mendeskripsikan kondisi dari data yang sudah anda miliki Dan menyajikannya dalam bentuk tabel diagram grafik dan bentuk lainnya yang disajikan dalam uraian-uraian singkat dan terbatas.

<h2>Quiz 6 : Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [0]:
# code here
path = 'diamonds-1.csv'
df = pd.read_csv(path)
df.describe()

Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000      2.910000  
50%        5.700000      5.710000      3.530000  
75%        6.540000      6.540000      4.040000  
max       10.740000     58.900000     31.800000

- Nilai rata-rata dari column price adalah : 3932.799722
- Nilai standar deviasa dari column depth adalah : 1.432621
- Nilai maximum dari column carat adalah : 5.01000